Compatibility with Boost 1.77 and upcoming Boost 1.78

Opening a more detailed issue in favor of #12732 : It is currently not possible to compile against boost 1.77 and 1.78 with (at least) the following three major issues:

(a) The following code in https://github.com/dealii/dealii/blob/master/source/grid/grid_tools.cc#L5634
```
const auto leaf = b_tree.qbegin(bgi::nearest(points[i], 1));
```
triggers
```
.../boost/geometry/algorithms/detail/comparable_distance/interface.hpp:122:21: error: ‘apply’ is not a member of ‘boost::geometry::dispatch::distance<dealii::Point<1>, boost::geometry::model::box<boost::geometry::model::point<double, 1, boost::geometry::cs::cartesian> >, boost::geometry::strategies::distance::detail::comparable<boost::geometry::strategies::distance::cartesian<> >, boost::geometry::point_tag, boost::geometry::box_tag, boost::geometry::strategy_tag_distance_point_box, false>’
```

(b) Just trying to compile `grid_tools.cc` fails with:
```
[... static assert triggers because of: ...]
.../boost/geometry/algorithms/not_implemented.hpp: In instantiation of ‘struct boost::geometry::nyi::not_implemented_error<boost::geometry::point_tag
, boost::geometry::box_tag>’
[... due to: ...]
.../boost/geometry/algorithms/dispatch/distance.hpp:123:8:   required from ‘struct boost::geometry::dispatch::distance<dealii::Point<1>, boost::geometry::model::box<boost::geometry::model::point<double, 1, boost::geometry::cs::cartesian> >, boost::geometry::strategies::distance::detail::comparable<boost::geometry::strategies::distance::cartesian<> >, boost::geometry::point_tag, boost::geometry::box_tag, boost::geometry::strategy_tag_distance_point_box, false>’
```

So I am a bit at a loss here. Does this imply that we lost support for our bounding box implementation due to some internal restructuring? If yes, what shall we do? We will need support with newer versions of boost rather sooner than later.

--- a/cmake/configure/configure_2_boost.cmake
+++ b/cmake/configure/configure_2_boost.cmake
@@ -146,25 +146,6 @@ MACRO(FEATURE_BOOST_FIND_EXTERNAL var)
   IF(BOOST_FOUND)
     SET(${var} TRUE)
 
-    #
-    # Set BOOST_DIR to something meaningful if empty
-    #
-    IF("${BOOST_DIR}" STREQUAL "")
-      SET(BOOST_DIR "<system location>")
-    ENDIF()
-
-    IF(BOOST_VERSION VERSION_EQUAL 1.77)
-      MESSAGE(STATUS "Could not find a sufficient Boost installation: "
-        "deal.II is not compatible with Boost version 1.77."
-        )
-      SET(BOOST_ADDITIONAL_ERROR_STRING
-        ${BOOST_ADDITIONAL_ERROR_STRING}
-        "The Boost installation (found at \"${BOOST_DIR}\")\n"
-        "with version ${BOOST_VERSION} is not compatible with deal.II.\n\n"
-        )
-      SET(${var} FALSE)
-    ENDIF()
-
     IF(DEAL_II_WITH_ZLIB)
       #
       # Test that Boost.Iostreams is usable.

--- a/include/deal.II/numerics/rtree.h
+++ b/include/deal.II/numerics/rtree.h
@@ -26,6 +26,8 @@
 #include <deal.II/boost_adaptors/segment.h>
 
 DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+#include <boost/geometry/algorithms/distance.hpp>
+
 #ifdef DEAL_II_BOOST_HAS_BROKEN_HEADER_DEPRECATIONS
 #  define BOOST_ALLOW_DEPRECATED_HEADERS
 #endif

--- a/tests/boost/rtree_01.cc
+++ b/tests/boost/rtree_01.cc
@@ -19,6 +19,7 @@
 
 #include <deal.II/boost_adaptors/point.h>
 
+#include <boost/geometry/algorithms/distance.hpp>
 #include <boost/geometry/index/rtree.hpp>
 #include <boost/geometry/strategies/strategies.hpp>
 

